src/attr/_funcs.py

```diff
@@ -25,6 +26,11 @@ def asdict(inst, recurse=True, filter=None, dict_factory=dict):
         example, to produce ordered dictionaries instead of normal Python
         dictionaries, pass in ``collections.OrderedDict``.
 
+    :param bool retain_collection_types: Do not convert to ``list`` when
+        encountering an attribute which is type ``tuple`` or ``set``.
+        ``recurse`` must also be ``True``.
```

Comment:
I think this is misleading.   More like "Only meaningful if ``recurse`` is ``True``."?